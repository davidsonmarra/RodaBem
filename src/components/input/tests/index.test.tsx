import React from 'react';
import {create} from 'react-test-renderer';
import Input, {Props as InputProps} from '..';

let mockValue = '';
const mockOnChangeText = jest.fn((text: string) => {
  mockValue = text;
});

const createComponent = ({
  value = '',
  onChangeText = mockOnChangeText,
  ...rest
}: InputProps) =>
  create(<Input value={value} onChangeText={onChangeText} {...rest} />);

describe('Input', () => {
  beforeEach(() => {
    mockValue = '';
    mockOnChangeText.mockClear();
  });

  it('renders correctly', () => {
    const tree = createComponent({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should update the value when the user types', () => {
    const typedText = 'new value typed in the input';
    const instance = createComponent({}).root;
    instance.props.onChangeText(typedText);
    expect(mockValue).toBe(typedText);
  });

  it('should render input with the correct value for default value', () => {
    const value = 'default value';
    const instance = createComponent({value}).root;
    expect(instance.props.value).toBe(value);
  });

  it('should render input with the correct value for label prop', () => {
    const label = 'label';
    const instance = createComponent({label}).root;
    expect(instance.props.label).toBe(label);
  });

  it('should render input with the correct value for placeholder prop', () => {
    const placeholder = 'placeholder';
    const instance = createComponent({placeholder}).root;
    expect(instance.props.placeholder).toBe(placeholder);
  });
});
