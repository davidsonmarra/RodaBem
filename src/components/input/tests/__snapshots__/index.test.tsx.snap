// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#222222",
          "fontSize": 14,
          "fontWeight": "normal",
        },
        {
          "width": "100%",
        },
      ]
    }
  />
  <View
    style={
      {
        "height": 8,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#595959",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 62,
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <TextInput
      onChangeText={[MockFunction]}
      style={
        {
          "flex": 1,
        }
      }
      value=""
    />
  </View>
</View>
`;
